src/test/java/org/opensearch/neuralsearch/query/HybridQueryFilterIT.java

```diff
@@ -0,0 +1,354 @@
+    private static final int INTEGER_FIELD_STOCK_5_20 = 20;
+    private static final String KEYWORD_FIELD_CATEGORY_1_DRAMA = "Drama";
+    private static final String KEYWORD_FIELD_CATEGORY_2_ACTION = "Action";
+    private static final String KEYWORD_FIELD_CATEGORY_3_SCI_FI = "Sci-fi";
+    private static final int SHARDS_COUNT_IN_SINGLE_NODE_CLUSTER = 1;
+    private static final int LTE_OF_RANGE_IN_HYBRID_QUERY = 400;
+    private static final int GTE_OF_RANGE_IN_HYBRID_QUERY = 200;
+    private static final int LTE_OF_RANGE_IN_POST_FILTER_QUERY = 400;
+    private static final int GTE_OF_RANGE_IN_POST_FILTER_QUERY = 230;
+    private static final String TEST_KNN_VECTOR_FIELD_NAME_1 = "test-knn-vector-1";
+    private final float[] testVector1 = createRandomVector(TEST_DIMENSION);
+    private final float[] testVector2 = createRandomVector(TEST_DIMENSION);
+    private final float[] testVector3 = createRandomVector(TEST_DIMENSION);
+
+    @BeforeClass
+    @SneakyThrows
+    public static void setUpCluster() {
+        // we need new instance because we're calling non-static methods from static method.
+        // main purpose is to minimize network calls, initialization is only needed once
+        HybridQueryFilterIT instance = new HybridQueryFilterIT();
+        instance.initClient();
+        instance.updateClusterSettings();
+    }
+
+    @SneakyThrows
+    public void testFilterOnNeuralQueryFilterAndTermQueryFilter_thenSuccessful() {
+        updateClusterSettings(CONCURRENT_SEGMENT_SEARCH_ENABLED, true);
+        prepareResourcesBeforeTestExecution(SHARDS_COUNT_IN_SINGLE_NODE_CLUSTER);
+        testNeuralQueryBuilder(TEST_MULTI_DOC_INDEX_WITH_TEXT_AND_INT_SINGLE_SHARD);
+    }
+
+    @SneakyThrows
+    private void testNeuralQueryBuilder(String indexName) {
+        String modelId = null;
+        modelId = prepareModel();
+        NeuralQueryBuilder neuralQueryBuilderTextQuery = NeuralQueryBuilder.builder()
+            .fieldName(TEST_KNN_VECTOR_FIELD_NAME_1)
+            .queryText(TEST_QUERY_TEXT)
+            .modelId(modelId)
+            .k(1)
+            .build();
+
+        TermQueryBuilder termQueryBuilder = QueryBuilders.termQuery(TEXT_FIELD_1_NAME, TEST_QUERY_TEXT);
+
+        HybridQueryBuilder hybridQueryBuilder = new HybridQueryBuilder();
+        hybridQueryBuilder.add(neuralQueryBuilderTextQuery);
+        hybridQueryBuilder.add(termQueryBuilder);
+
+        hybridQueryBuilder.filter(new MatchQueryBuilder("_id", "1"));
+
+        Map<String, Object> searchResponseAsMapTextQuery = search(indexName, hybridQueryBuilder, 1);
+        assertEquals(1, getHitCount(searchResponseAsMapTextQuery));
+
+        Map<String, Object> firstInnerHitTextQuery = getFirstInnerHit(searchResponseAsMapTextQuery);
+        assertEquals("1", firstInnerHitTextQuery.get("_id"));
+    }
+
+    @SneakyThrows
+    public void testFilterOnIndexWithSingleShard_whenConcurrentSearchEnabled_thenSuccessful() {
```

Comment:
I don't think we need this match of tests for CSS, one should be enough, this one you can delete